### ISSUE TITLE
Does not detect DNS-based ad-blockers

### ISSUE THREAD

**Issue by stefanbohacek** [stefanbohacek/detect-missing-adblocker#20]
- [PiHole](https://pi-hole.net/)
- [AdGuard](https://adguard-dns.io/en/welcome.html)

These will probably require some additional JS code.

**Comment by thelovebug** [comment]
Please add [NextDNS](https://nextdns.io/) to this list 🙂

**Comment by stefanbohacek** [comment]
I think [ProtonVPN](https://protonvpn.com/) belongs here as well?

**Comment by stefanbohacek** [comment]
@cku-heise Yes, a previous version of the plugin relied on loading the `nativeads.js.css` file, which should be blocked by DNS-based ad-blockers, but that still didn't seem to work.

I am also wondering if this should perhaps be "intended behavior". The bottom line is, this plugin renders and "ad" in a way that's hard for DNS-based ad-blockers to detect and block. If I say to a user, you're good, you're blocking ads, is that perhaps a bit inaccurate, if I am in fact able to serve an ad that is easily detected and blocked by other tools?

**Comment by cku-heise** [comment]
That's a good point. A DNS-based blocker can, by design, not block all ads. It boils down to your motivation for creating the extension: Do you want to incentivize people to install software that blocks _all_ kinds of ads or just a relevant subset? In either case, maybe you could rephrase the verbiage to add a small sentence about DNS-based blockers. Something like "This extension is currently not designed to detect DNS-based ad blockers, this is a deliberate design decision", linking to a FAQ article or something. 
I still think your extension is a very interesting concept!

**Comment by stefanbohacek** [comment]
@cku-heise Thank you!

I went ahead and added a note to the README for now, will definitely continue exploring this.
